### PR TITLE
fix: disable forwarding node stamp verification

### PR DESCRIPTION
**Pull request by istae**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2233)

### CODE REVIEW

**Comment by zelig** [comment]
i think doing this in general is not a good idea. 
we should be lenient with validation only if we are lagging behind in blockchain sync
if we want to speed up forwarding then lets do spotchecks with fixed probability

**Review by acud — CHANGES_REQUESTED** [review]
> Reviewed 4 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2233)* status: 4 of 5 files reviewed, 2 unresolved discussions (waiting on @aloknerurkar, @esadakar, and @metacertain)

___
*[pkg/postage/stamp.go, line 117 at r1](https://reviewable.io/reviews/ethersphere/bee/2233#-MdSJezK8PgSduw8KmtH:-MdSJezK8PgSduw8KmtI:bb4gpad) ([raw file](https://github.com/ethersphere/bee/blob/a7e54af039b5ea4d82cdaf83ef01482ec97c55aa/pkg/postage/stamp.go#L117)):*
> ```Go
> 
> // ValidStampBytes returns a stampvalidator function passed to protocols with chunk entrypoints.
> func ValidStampBytes(batchStore Storer) func(chunk swarm.Chunk, stampBytes []byte) (swarm.Chunk, error) {
> ```

not sure - why do we need both methods? one would do, and both actually do the complete stamp validation (as the second one tries to find the stamp in the batchstore too). 
___
*[pkg/pushsync/pushsync.go, line 275 at r1](https://reviewable.io/reviews/ethersphere/bee/2233#-MdSKN8E5nMmYyVyhyuf:-MdSKN8E5nMmYyVyhyug:b3jirk7) ([raw file](https://github.com/ethersphere/bee/blob/a7e54af039b5ea4d82cdaf83ef01482ec97c55aa/pkg/pushsync/pushsync.go#L275)):*
> ```Go
> }
> 
> func (ps *PushSync) validateStamp(ch swarm.Chunk) (swarm.Chunk, error) {
> ```

just a suggestion, but since this just wraps the `validStamp` call and since the calling context checks the error that is returned here, maybe its just as good to reduce indirection and do the explicit `validStamp` wherever needed and instrument the returned error in a similar manner?

File: pkg/pushsync/pushsync.go [hunk 1]
```diff
@@ -139,9 +140,13 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 	ps.metrics.TotalReceived.Inc()
 
 	chunk := swarm.NewChunk(swarm.NewAddress(ch.Address), ch.Data)
-	if chunk, err = ps.validStamp(chunk, ch.Stamp); err != nil {
-		return fmt.Errorf("pushsync valid stamp: %w", err)
+	chunkAddress := chunk.Address()
```
Comment (aloknerurkar) [inline pkg/pushsync/pushsync.go:143]: I guess setting the stamp here is not required if you use the new validStampBytes call which would do this.

File: pkg/node/node.go [hunk 2]
```diff
@@ -402,6 +402,7 @@ func NewBee(addr string, publicKey *ecdsa.PublicKey, signer crypto.Signer, netwo
 	unreserveFn = storer.UnreserveBatch
 
 	validStamp := postage.ValidStamp(batchStore)
```
Comment (aloknerurkar) [inline pkg/node/node.go:404]: I think it would be better to remove this. Use the bytes handler everywhere?

**Review by esadakar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2233)* status: 4 of 5 files reviewed, 4 unresolved discussions (waiting on @acud, @aloknerurkar, @esadakar, and @metacertain)

___
*[pkg/postage/stamp.go, line 117 at r1](https://reviewable.io/reviews/ethersphere/bee/2233#-MdSJezK8PgSduw8KmtH:-MdWEInW78nntViFkAgv:b-h2xjzd) ([raw file](https://github.com/ethersphere/bee/blob/a7e54af039b5ea4d82cdaf83ef01482ec97c55aa/pkg/postage/stamp.go#L117)):*
<details><summary><i>Previously, acud (acud) wrote…</i></summary><blockquote>

not sure - why do we need both methods? one would do, and both actually do the complete stamp validation (as the second one tries to find the stamp in the batchstore too). 
</blockquote></details>

With this PR, handler method at the beginning Unmarshalls the stamp without validating it, so it felt unnecessary to call ValidStamp with the bytes value again since it was already parsed.  
___
*[pkg/pushsync/pushsync.go, line 143 at r1](https://reviewable.io/reviews/ethersphere/bee/2233#-MdHfMl2TC0yh_C9c5CP-r1-143:-MdWE3wC81gLz69RGX36:b-l0gjeg) ([raw file](https://github.com/ethersphere/bee/blob/a7e54af039b5ea4d82cdaf83ef01482ec97c55aa/pkg/pushsync/pushsync.go#L143)):*
<details><summary><i>Previously, aloknerurkar wrote…</i></summary><blockquote>

I guess setting the stamp here is not required if you use the new validStampBytes call which would do this.
</blockquote></details>

The point is to not validate it at the beginning of the handler.
___
*[pkg/pushsync/pushsync.go, line 275 at r1](https://reviewable.io/reviews/ethersphere/bee/2233#-MdSKN8E5nMmYyVyhyuf:-MdWFQIy1G6HBg_ZHoP4:b2nhll9) ([raw file](https://github.com/ethersphere/bee/blob/a7e54af039b5ea4d82cdaf83ef01482ec97c55aa/pkg/pushsync/pushsync.go#L275)):*
<details><summary><i>Previously, acud (acud) wrote…</i></summary><blockquote>

just a suggestion, but since this just wraps the `validStamp` call and since the calling context checks the error that is returned here, maybe its just as good to reduce indirection and do the explicit `validStamp` wherever needed and instrument the returned error in a similar manner?
</blockquote></details>

Ok.

**Review by esadakar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2233)* status: 3 of 5 files reviewed, 4 unresolved discussions (waiting on @acud, @aloknerurkar, @esadakar, and @metacertain)

___
*[pkg/postage/stamp.go, line 117 at r1](https://reviewable.io/reviews/ethersphere/bee/2233#-MdSJezK8PgSduw8KmtH:-MdWOaT0ENMqP8tMGPx-:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/a7e54af039b5ea4d82cdaf83ef01482ec97c55aa/pkg/postage/stamp.go#L117)):*
<details><summary><i>Previously, esadakar (Esad) wrote…</i></summary><blockquote>

With this PR, handler method at the beginning Unmarshalls the stamp without validating it, so it felt unnecessary to call ValidStamp with the bytes value again since it was already parsed.  
</blockquote></details>

Done.

**Review by acud — APPROVED** [review]
> Reviewed 5 of 5 files at r2.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2233)* status: all files reviewed, 2 unresolved discussions (waiting on @aloknerurkar, @esadakar, and @metacertain)

___
*[pkg/pushsync/pushsync_test.go, line 810 at r2](https://reviewable.io/reviews/ethersphere/bee/2233#-MdXRhKu-w_THAZDzGko:-MdXRhKu-w_THAZDzGkp:b-d51367) ([raw file](https://github.com/ethersphere/bee/blob/baba48564ecbc61b3e6f5d2a0a2af9dd66edffa7/pkg/pushsync/pushsync_test.go#L810)):*
> ```Go
> 	}
> 
> 	validStamp := func(ch swarm.Chunk, stapp []byte) (swarm.Chunk, error) {
> ```

stamp :)

**Review by esadakar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2233)* status: all files reviewed, 2 unresolved discussions (waiting on @aloknerurkar, @esadakar, and @metacertain)

___
*[pkg/pushsync/pushsync_test.go, line 810 at r2](https://reviewable.io/reviews/ethersphere/bee/2233#-MdXRhKu-w_THAZDzGko:-MdXb8vY6zHk3ST5_mGH:b32vhv) ([raw file](https://github.com/ethersphere/bee/blob/baba48564ecbc61b3e6f5d2a0a2af9dd66edffa7/pkg/pushsync/pushsync_test.go#L810)):*
<details><summary><i>Previously, acud (acud) wrote…</i></summary><blockquote>

stamp :)
</blockquote></details>

:(

**Review by acud — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2233#-:-MdasA-k3gGSH9itVxID:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @aloknerurkar, @istae, and @metacertain)

**Review by aloknerurkar — APPROVED** [review]

**Pull request merged**